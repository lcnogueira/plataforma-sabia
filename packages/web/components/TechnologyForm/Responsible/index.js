/* eslint-disable no-use-before-define */
import React from 'react';
import PropTypes from 'prop-types';
import { FaPlus, FaMinus } from 'react-icons/fa';
import { InputField } from '../../Form';
import { Wrapper } from './styles';
import { CircularButton } from '../../Button';
import Repeater from '../../Form/Repeater';
import { useAuth } from '../../../hooks';
import { Cell, Row } from '../../Common/Layout';

const Responsible = ({ form }) => {
	const { user } = useAuth();
	const emptyValue = {
		fullName: '',
		email: '',
		phoneNumber: '',
		lattesID: '',
	};

	return (
		<Wrapper>
			<Row>
				<Cell col={5}>
					<InputField
						form={form}
						name="technologyResponsibles.owner.full_name"
						label="Nome Completo"
						disabled
						defaultValue={user.full_name}
					/>
				</Cell>
				<Cell col={3}>
					<InputField
						form={form}
						name="technologyResponsibles.owner.email"
						label="Email"
						disabled
						defaultValue={user.email}
					/>
				</Cell>
				<Cell col={2}>
					<InputField
						form={form}
						name="technologyResponsibles.owner.phone_number"
						label="Telefone"
						placeholder="(xx) xxxx - xxxx"
						disabled
						defaultValue={user.phone_number}
					/>
				</Cell>
				<Cell col={2}>
					<InputField
						form={form}
						name="technologyResponsibles.owner.lattes_id"
						label="ID Lattes"
						placeholder="Somente números"
						validation={{ required: true }}
						defaultValue={user.lattes_id}
					/>
				</Cell>
				<Cell maxWidth={0.5} />
			</Row>
			<Repeater
				form={form}
				name="responsible"
				emptyValue={emptyValue}
				childsComponent={({ item, index, remove, fields }) => {
					return (
						<>
							<Row key={item.id} align="center">
								<Cell col={5}>
									<InputField
										form={form}
										name={`technologyResponsibles.users[${index}].full_name`}
										label="Nome Completo"
										placeholder="Nome do responsável"
										validation={{ required: true }}
									/>
								</Cell>
								<Cell col={3}>
									<InputField
										form={form}
										name={`technologyResponsibles.users[${index}].email`}
										label="Email"
										placeholder="Ex.: email@dominio.com.br"
										validation={{ required: true }}
									/>
								</Cell>
								<Cell col={2}>
									<InputField
										form={form}
										name={`technologyResponsibles.users[${index}].phone_number`}
										label="Telefone"
										placeholder="(xx) xxxx - xxxx"
										validation={{ required: true }}
									/>
								</Cell>
								<Cell col={2}>
									<InputField
										form={form}
										name={`technologyResponsibles.users[${index}].lattes_id`}
										label="ID Lattes"
										placeholder="Somente números"
										validation={{ required: true }}
									/>
								</Cell>

								<Cell maxWidth={0.5}>
									<CircularButton
										disabled={index === 0 && fields.length === 1}
										size="small"
										variant="remove"
										shortPadding
										onClick={(event) => {
											event.preventDefault();
											remove(index);
										}}
									>
										<FaMinus />
									</CircularButton>
								</Cell>
							</Row>
						</>
					);
				}}
				// eslint-disable-next-line no-shadow
				endComponent={({ append, emptyValue }) => {
					return (
						<CircularButton
							right
							variant="info"
							color="white"
							onClick={(event) => {
								event.preventDefault();
								append(emptyValue);
							}}
						>
							<FaPlus />
						</CircularButton>
					);
				}}
			/>
		</Wrapper>
	);
};

Responsible.propTypes = {
	form: PropTypes.shape({}),
};

Responsible.defaultProps = {
	form: {},
};

export default Responsible;
