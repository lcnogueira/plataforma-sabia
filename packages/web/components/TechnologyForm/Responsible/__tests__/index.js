/* eslint-disable jest/no-disabled-tests */
import React from 'react';
import { render, fireEvent, screen } from 'test-utils';
import { useForm } from 'react-hook-form';
import Responsible from '..';

jest.mock('react-icons/fa', () => {
	return {
		__esModule: true,
		FaPlus: () => {
			return <svg>FaPlus mocked</svg>;
		},
		FaMinus: () => {
			return <svg>FaMinus mocked</svg>;
		},
	};
});

// eslint-disable-next-line react/prop-types
const ResponsibleWrapper = () => {
	const onSubmit = jest.fn();

	const form = useForm({ mode: 'onChange' });

	return (
		<form onSubmit={form.handleSubmit(onSubmit)}>
			<Responsible form={form} />
		</form>
	);
};

test('it renders the form with one row', () => {
	const { container } = render(<ResponsibleWrapper />);

	expect(container).toMatchSnapshot();
});

test('it increase the number of rows when clicking in add button', () => {
	render(<ResponsibleWrapper />);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[1]);

	expect(buttons[1]).toMatchSnapshot();
});

test('it decrease the number of rows when clicking in remove button', () => {
	const { container } = render(<ResponsibleWrapper />);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[1]);

	const buttons2 = screen.getAllByRole('button');

	fireEvent.click(buttons2[0]);

	expect(container).toMatchSnapshot();
});

test('it should not remove the row when only one is available', () => {
	const { container } = render(<ResponsibleWrapper />);

	const buttons = screen.getAllByRole('button');

	fireEvent.click(buttons[0]);

	expect(container).toMatchSnapshot();
});
