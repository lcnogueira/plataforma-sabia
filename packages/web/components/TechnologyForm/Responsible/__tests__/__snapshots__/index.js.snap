// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it decrease the number of rows when clicking in remove button 1`] = `
.c4 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c7 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c7::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c10 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c9 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9:hover {
  opacity: 0.8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c3 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  margin-bottom: 4rem;
}

.c1 {
  margin-bottom: 3rem;
}

@media (max-width:450px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <form>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
            size="2"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].fullName"
              >
                Nome Completo
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Nome Completo"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].fullName"
                  name="responsible[0].fullName"
                  placeholder="Nome do responsável"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].email"
              >
                Email
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Email"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].email"
                  name="responsible[0].email"
                  placeholder="Ex.: email@dominio.com.br"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].phone"
              >
                Telefone
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Telefone"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].phone"
                  name="responsible[0].phone"
                  placeholder="(xx) xxxx - xxxx"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].lattesId"
              >
                ID Lattes
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="ID Lattes"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].lattesId"
                  name="responsible[0].lattesId"
                  placeholder="Somente números"
                  type="text"
                />
              </div>
            </div>
          </div>
          <button
            class="c9"
            color="hsla(0, 0%, 100%)"
            disabled=""
            name=""
            title=""
            type="button"
          >
            <svg>
              FaMinus mocked
            </svg>
          </button>
        </div>
        <button
          class="c10"
          color="hsla(0, 0%, 100%)"
          name=""
          title=""
          type="button"
        >
          <svg>
            FaPlus mocked
          </svg>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`it increase the number of rows when clicking in add button 1`] = `
.c0 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0:hover {
  opacity: 0.8;
}

@media (max-width:450px) {

}

@media (max-width:450px) {

}

@media (max-width:450px) {

}

<button
  class="c0"
  color="hsla(0, 0%, 100%)"
  name=""
  title=""
  type="button"
>
  <svg>
    FaPlus mocked
  </svg>
</button>
`;

exports[`it renders the form with one row 1`] = `
.c4 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c7 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c7::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c10 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c9 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9:hover {
  opacity: 0.8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c3 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  margin-bottom: 4rem;
}

.c1 {
  margin-bottom: 3rem;
}

@media (max-width:450px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <form>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
            size="2"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].fullName"
              >
                Nome Completo
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Nome Completo"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].fullName"
                  name="responsible[0].fullName"
                  placeholder="Nome do responsável"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].email"
              >
                Email
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Email"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].email"
                  name="responsible[0].email"
                  placeholder="Ex.: email@dominio.com.br"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].phone"
              >
                Telefone
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Telefone"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].phone"
                  name="responsible[0].phone"
                  placeholder="(xx) xxxx - xxxx"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].lattesId"
              >
                ID Lattes
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="ID Lattes"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].lattesId"
                  name="responsible[0].lattesId"
                  placeholder="Somente números"
                  type="text"
                />
              </div>
            </div>
          </div>
          <button
            class="c9"
            color="hsla(0, 0%, 100%)"
            disabled=""
            name=""
            title=""
            type="button"
          >
            <svg>
              FaMinus mocked
            </svg>
          </button>
        </div>
        <button
          class="c10"
          color="hsla(0, 0%, 100%)"
          name=""
          title=""
          type="button"
        >
          <svg>
            FaPlus mocked
          </svg>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`it should not remove the row when only one is available 1`] = `
.c4 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c7 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c7::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c7::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c10 {
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c10:hover {
  opacity: 0.8;
}

.c9 {
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9:hover {
  opacity: 0.8;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > *:not(:first-child):not(:last-child) {
  margin: 0 10px;
}

.c3 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  margin-bottom: 4rem;
}

.c1 {
  margin-bottom: 3rem;
}

@media (max-width:450px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-top: 1.5rem;
  }
}

@media (max-width:450px) {
  .c3 {
    width: 100%;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<div>
  <form>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <div
            class="c3"
            size="2"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].fullName"
              >
                Nome Completo
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Nome Completo"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].fullName"
                  name="responsible[0].fullName"
                  placeholder="Nome do responsável"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].email"
              >
                Email
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Email"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].email"
                  name="responsible[0].email"
                  placeholder="Ex.: email@dominio.com.br"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].phone"
              >
                Telefone
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="Telefone"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].phone"
                  name="responsible[0].phone"
                  placeholder="(xx) xxxx - xxxx"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="c8"
          >
            <div
              class="c4"
            >
              <label
                class="c5"
                for="responsible[0].lattesId"
              >
                ID Lattes
              </label>
              <div
                class="c6"
              >
                <input
                  aria-label="ID Lattes"
                  aria-required="true"
                  class="c7"
                  id="responsible[0].lattesId"
                  name="responsible[0].lattesId"
                  placeholder="Somente números"
                  type="text"
                />
              </div>
            </div>
          </div>
          <button
            class="c9"
            color="hsla(0, 0%, 100%)"
            disabled=""
            name=""
            title=""
            type="button"
          >
            <svg>
              FaMinus mocked
            </svg>
          </button>
        </div>
        <button
          class="c10"
          color="hsla(0, 0%, 100%)"
          name=""
          title=""
          type="button"
        >
          <svg>
            FaPlus mocked
          </svg>
        </button>
      </div>
    </div>
  </form>
</div>
`;
